Modules/_csv.c

```diff
@@ -570,13 +577,17 @@ parse_grow_buff(ReaderObj *self)
 static int
 parse_add_char(ReaderObj *self, Py_UCS4 c)
 {
-    if (self->field_len >= _csvstate_global->field_limit) {
-        PyErr_Format(_csvstate_global->error_obj, "field larger than field limit (%ld)",
-                     _csvstate_global->field_limit);
+    PyTypeObject *reader_type = Py_TYPE(self);
+    _csvstate *state = PyType_GetModuleState(reader_type);
```

Comment:
The Reader type has Py_TPFLAGS_BASETYPE: PyType_GetModuleState() is unsafe here.